test/src/tests/generated/paging/models.go

```diff
@@ -185,6 +190,11 @@ func (page OdataProductResultPage) Values() []Product {
 	return *page.opr.Values
 }
 
+// Creates a new instance of the OdataProductResultPage type.
+func NewOdataProductResultPage(getNextPage func(context.Context, OdataProductResult) (OdataProductResult, error)) OdataProductResultPage {
```

Comment:
hmm, can we create a function type , instead of inlining it ?

Reply:
So I didn't because this is the only place where the type is publicly exposed and I didn't want to add more stuff to the namespace.  Are you worried about the length?

Reply:
hmm if its the only place used, I guess its fine... only thing would be if they wan to pass the function around through their code they would need to define themselves the type or just have the param as we do

Reply:
Good point.  I suggest we leave it as-is and if we get a complaint/ask then we make that change.